anesthetic/utils.py

```diff
@@ -335,28 +335,22 @@ def sample_compression_1d(x, w=None, n=1000):
     x, w, array-like
         Compressed samples and weights
     """
-    x = pandas.Series(x)
+    x = np.array(x)
     if w is None:
-        w = pandas.Series(index=x.index, data=np.ones_like(x))
+        w = np.ones_like(x)
+    w = np.array(w)
 
     # Select inner samples for triangulation
-    if sum(w != 0) < n:
-        i = w.index
-    else:
-        i = np.random.choice(w.index, size=n, replace=False, p=w/w.sum())
-
-    # Define sub-samples
-    x_ = np.sort(x[i])
+    x_ = x.copy()
+    if len(x_) > n:
+        x_ = np.random.choice(x_, size=n, replace=False)
+    x_ = np.sort(x_)
```

Comment:
Do we need to make a copy of `x` here? Couldn't we just do `np.random.choice(x, ...)`?

Reply:
Yes. We need x_ (which represents the 'bins' that we round to) and x (to round in np.digitize).

Reply:
I meant to do `x_ = np.random.choice(x, ...)` and only do `x_ = x.copy()` if necessary.

Granted, this might be evil premature optimisation...

Reply:
I think in this case the sort is much more expensive than the copy, so its definitely a case of unecessary optimisation. The new version is an extra line of code, so the better question is which is clearer pythonically?

Reply:
I think yours, so let's go with that.